examples/technical_blogs/tartan_blog/build_and_run_networks.py

```diff
@@ -0,0 +1,187 @@
+        "-GNinja",
+        f"-DCMAKE_CXX_COMPILER={os.environ['CMAKE_CXX_COMPILER']}",
+        "-DSNN_BUILD_BENCHMARKS=OFF",
+        "-DSNN_BENCH_SYCLBLAS=OFF",
+        "-DSNN_BUILD_DOCUMENTATION=OFF",
+        f"-DSyclBLAS_DIR='{os.path.join(script_directory, 'build_portBlas')}'"
+    ]
+
+    run_cmake_and_ninja(cmake_sycl_dnn_command, sycl_dnn_build_dir)
+
+
+def download_and_convert_model(dir, file_url, destination_file, h5_to_bin_script):
+    #Check if the model dir already exists
+    path = os.path.join(os.getcwd(), dir)
+    if os.path.exists(path):
+        print(f"'{path}' already exists. Skipping creating '{path}'.")
+    else:
+        os.mkdir(path)
+        print(f"Directory '{path}' created successfully.")
+
+    # Check if the file already exists
+    if os.path.exists(destination_file):
+        print(f"'{destination_file}' already exists. Skipping download.")
+    else:
+        try:
+            response = requests.get(file_url)
+            response.raise_for_status()
+            with open(destination_file, "wb") as file:
+                file.write(response.content)
+            print(f"Downloaded '{destination_file}' successfully.")
+        
+            subprocess.run(["python", h5_to_bin_script, destination_file], check=True)
+            print(f"Converted '{destination_file}' to binary successfully.")
+            os.chdir("..")  # Move up one level to the parent directory
+        except requests.exceptions.RequestException as e:
+            print(f"Error downloading file: {e}", file=sys.stderr)
+            sys.exit(1)
+        except subprocess.CalledProcessError as e:
+            print(f"Error running h5toBin.py: {e}", file=sys.stderr)
+            sys.exit(1)
+
+
+def prepare_image(img_script_path, input_image):
+    try:
+        subprocess.run(["python", img_script_path, input_image], check=True)
+        print(f"Image prepared successfully.")
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing the script: {e}", file=sys.stderr)
+        sys.exit(1)
+
+# Test on images
+def test_image(model_script, data_dir, output_file):
+    command_args = [
+        model_script,
+        data_dir + "/",
+        output_file
+    ]
+    try:
+        subprocess.run(command_args, check=True)
+        print(f"Image tested successfully with {os.path.basename(model_script)}.")
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing the command: {e}", file=sys.stderr)
+        sys.exit(1)
+
+
+def main():
+    build_portBLAS()
+    build_sycl_dnn()
+    sycl_dnn_build_dir = os.path.join(os.getcwd(), "build_sycl_dnn")
+    vgg_dir = os.path.join(os.getcwd(), "vdata")
+    resnet_dir = os.path.join(os.getcwd(), "rdata")
+
+    # Set the environment variables
+    os.environ["CA_HAL_DEBUG"] =  os.environ.get("CA_HAL_DEBUG", "1")
+    os.environ["CA_PROFILE_LEVEL"] = os.environ.get("CA_PROFILE_LEVEL", "3")
+    os.environ["ONEAPI_DEVICE_SELECTOR"] = "opencl::acc"
```

Comment:
might be worth considering allowing the user to add what they want, we may be one of many accelerators